NXP drivers: i2c_mcux_flexcomm: adds PM TURN_ON low-power recovery support

DerekSnell opened the pull request:
Enables Sleep mode (PM3) in RW61x.

Tested samples/sensor/thermometer on FRDM-RW612, I2C transfers continue every 1s, entering PM3 mode in between readings.

Commit:
drivers: i2c_mcux_flexcomm: adds PM TURN_ON low-power recovery support

Enables Sleep mode (PM3) in RW61x.

Signed-off-by: Derek Snell <derek.snell@nxp.com>

mmahadevan108 commented:
@DerekSnell, it would be good to enable PM Mode 3 by default for RW612 boards via board specific overlays.

mmahadevan108 commented on drivers/i2c/i2c_mcux_flexcomm.c:
```diff
@@ -25,6 +25,7 @@ LOG_MODULE_REGISTER(mcux_flexcomm);
 
+#include <zephyr/pm/device.h>
 #define I2C_TRANSFER_TIMEOUT_MSEC                                                                  \
 	COND_CODE_0(CONFIG_I2C_NXP_TRANSFER_TIMEOUT, (K_FOREVER),                                  \
```
I find it strange that we do not have to block off entering low power modes during I2C transfers.

Commit:
samples: sensor: thermometer: frdm_rw612: enable Standby power mode

Enables System PM and Standby (AKA Sleep or PM3) mode, to test I2C
after waking from Standby mode.

Signed-off-by: Derek Snell <derek.snell@nxp.com>

mmahadevan108 commented:
@teburd , we have tested this change on our boards with the thermometer sample

teburd commented:
Thank you for re-assigning, unless this is a new driver or implementing completely new functionality my assignment and/or +1 is not required and should not be required.
